### PR TITLE
Don't require AWS access keys for S3 pytests

### PR DESCRIPTION
**Pull request by arpad-m**
Don't require AWS access keys (AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY) for S3 usage in the pytests, and also allow AWS_PROFILE to be passed.

One of the two methods is required however.

This allows local development like:

```
aws sso login --profile dev
export ENABLE_REAL_S3_REMOTE_STORAGE=nonempty REMOTE_STORAGE_S3_REGION=eu-central-1 REMOTE_STORAGE_S3_BUCKET=neon-github-ci-tests AWS_PROFILE=dev
cargo build_testing && RUST_BACKTRACE=1 ./scripts/pytest -k debug-pg16 test_runner/regress/test_tenant_delete.py::test_tenant_delete_smoke
```

related earlier PR for the cargo unit tests of the `remote_storage` crate: #6202

### CODE REVIEW

**Comment by arpad-m** [comment]
sadly it doesn't work however, the test fails and the pageserver log says it's an authentication issue:

```
2024-01-31T16:51:13.730780Z  INFO Using s3 bucket 'neon-github-ci-tests' in region 'eu-central-1' as a remote storage, prefix in bucket: 'Some("929215e1-150a-4671-a5f0-fa8511a54bc0/test-tenant-delete-smoke-debug-pg16-/pageserver")', bucket endpoint: 'None', profile: Some("dev"), access_key_id: None
[...]
2024-01-31T16:51:16.558695Z  INFO attach_preload{tenant_id=354d6017ac6faba459d228e056d30cd1 shard_id=0000}: list timelines for 354d6017ac6faba459d228e056d30cd1 failed, will retry (attempt 1): Failed to download a remote file: Failed to list S3 prefixes

Caused by:
    0: dispatch failure
    1: other
    2: the credential provider was not enabled
    3: no providers in chain provided credentials
```

Apparently the key does get picked up by the pageserver but for some reason it still doesn't pick up the auth information...

**Comment by arpad-m** [comment]
found the issue, had to also import the HOME env variable. See the new commit.

It works now locally, although i had to increase the iteration number, probably due to higher latencies from my box to S3:

```diff
diff --git a/test_runner/regress/test_tenant_delete.py b/test_runner/regress/test_tenant_delete.py
index b4e5a550f..392b8add6 100644
--- a/test_runner/regress/test_tenant_delete.py
+++ b/test_runner/regress/test_tenant_delete.py
@@ -89,7 +89,7 @@ def test_tenant_delete_smoke(
     iterations = poll_for_remote_storage_iterations(remote_storage_kind)
 
     assert ps_http.get_metric_value("pageserver_tenant_manager_slots") == 2
-    tenant_delete_wait_completed(ps_http, tenant_id, iterations)
+    tenant_delete_wait_completed(ps_http, tenant_id, iterations * 10)
     assert ps_http.get_metric_value("pageserver_tenant_manager_slots") == 1
 
     tenant_path = env.pageserver.tenant_dir(tenant_id)
```

**Review by bayandin — APPROVED** [review]
Nice! It's going to be much more convenient to use it locally

File: control_plane/src/background_process.rs [hunk 1]
```diff
@@ -256,6 +256,9 @@ fn fill_remote_storage_secrets_vars(mut cmd: &mut Command) -> &mut Command {
     for env_key in [
         "AWS_ACCESS_KEY_ID",
         "AWS_SECRET_ACCESS_KEY",
+        "AWS_PROFILE",
+        // HOME is needed in combination with `AWS_PROFILE` to pick up the SSO sessions.
+        "HOME",
         "AWS_SESSION_TOKEN",
```
[the remark this answers is not in the document]
Comment (arpad-m) [reply control_plane/src/background_process.rs:262]: We still have some support for that environment variable in the `S3Storage` fixture. I'd keep it around but if you really want I'll remove it.

[the remark this answers is not in the document]
Comment (bayandin) [reply control_plane/src/background_process.rs:262]: Wasn't it removed here https://github.com/neondatabase/neon/pull/6556/files#diff-49937a3aeaddc012b2cdd2585f05f37338c1e17bc438a77200b82e898471744aL324-L325?

I don't see if it is in use anywhere explicitly. In the code, there's only one line here:
```
$ git grep AWS_SESSION_TOKEN
control_plane/src/background_process.rs:        "AWS_SESSION_TOKEN",
```

**Pull request merged**